default/scripting/buildings/XENORESURRECTION_LAB.focs.txt

```diff
@@ -13,6 +13,26 @@ BuildingType
         ]
     ]
     EnqueueLocation = [[ENQUEUE_BUILD_ONE_PER_PLANET]]
+    effectsgroups = [
+        EffectsGroup
+            scope = And [
+                Source
+                ContainedBy HasSpecial name = "EXTINCT_BANFORO_SPECIAL"
```

Comment:
Activation conditions test only the source object (and not the planet a source building is on), so you need to check something like

    activation = ContainedBy And [
        Object id = Source.PlanetID
        HasSpecial name = "EXTINCT_BANFORO_SPECIAL"
    ]